### ISSUE TITLE
First WWIVnet connection problem

### ISSUE THREAD

**Issue by Thaeland** [wwivbbs/wwiv#1024]
I installed WWIV 5.2 on Ubuntu Linux 4.4.0-64 and installed wwivnet-75.zip detailed in the Linux instructions.  After running the network connection detailed in the installation for WWIVnet I am seeing this error:

./network --net=wwivnet --node=1
2017-02-25 16:31:19,585 INFO  NETWORK for network: wwivnet
2017-02-25 16:31:19,587 INFO  USE networkb: bbs.weather-station.org:24556
2017-02-25 16:31:19,587 INFO  Executing Command: '/home/wwiv/networkb --send --net=0 --node=1'
2017-02-25 16:31:19,595 INFO  BinkP send to: 1
2017-02-25 16:31:19,735 INFO  wwivnet_node_number_from_ftn_address: '20000:20000/1@wwivnet'
2017-02-25 16:31:19,837 INFO  OPT line: 'CRAM-MD5-cafebabecafebabecafebabecafebabe'
2017-02-25 16:31:19,837 INFO         CRAM Requested by Remote Side.
2017-02-25 16:31:19,838 INFO  Remote Side: Weather Station
2017-02-25 16:31:19,838 INFO  Remote Side: networkb/5.3.0.2481 binkp/1.0
2017-02-25 16:31:19,839 INFO  OPT line: 'CRC'
2017-02-25 16:31:19,839 INFO         Not enabling CRC support (disabled in net.ini).
2017-02-25 16:31:22,056 INFO  SEND:  M_NUL: WWIVVER 5.2.0.2430 (Dec 30 2016, 16:51:56)
2017-02-25 16:31:22,056 INFO  SEND:  M_NUL: SYS Sagramour's Realm
2017-02-25 16:31:22,057 INFO  SEND:  M_NUL: ZYZ Jon Beets aka Thaeland Sagramour
2017-02-25 16:31:22,058 INFO  SEND:  M_NUL: VER networkb/5.2.0.2430 binkp/1.0
2017-02-25 16:31:22,058 INFO  SEND:  M_NUL: LOC Unknown
2017-02-25 16:31:22,059 INFO  SEND:  M_NUL: WWIV @108.wwivnet
2017-02-25 16:31:22,059 INFO  SEND:  M_ADR: 20000:20000/108@wwivnet
2017-02-25 16:31:22,262 INFO  SEND:  M_PWD: ********
2017-02-25 16:31:22,769 ERROR M_ERR: Error (NETWORKB-0002): Unexpected Address: 20000:20000/108@wwivnet
2017-02-25 16:31:53,019 ERROR        after WaitOk: M_OK never received.
2017-02-25 16:31:53,019 INFO  SEND:  M_ERR: M_OK never received. Timeed out waiting for it.
2017-02-25 16:31:53,020 ERROR STATE: Error Received.

Jon Beets

**Comment by wwiv** [comment]
Usually that means either the remote (Mark) doesn't have you in his callout.net
https://github.com/wwivbbs/wwiv/blob/8c460f502ab64640c97e3b518d0dc26eea2451ec/networkb/binkp.cpp#L597 is where that errors is generated.

@MarkHofmann11 Any thoughts>

**Comment by MarkHofmann11** [comment]
Just verified that I do have @108 in my callout.net - complete with
password in quotes, etc.  I forced a network update and everything checks
out on this side (no errors in the feedback).

Will need to verify the callout.net data on the sender side.

**Comment by Thaeland** [comment]
I already sent you an email but my callout.net is below:

@1 & "mypassword"

Of course the password is the one you sent me with the quotes.

Jon

**Comment by granitepenguin** [comment]
try @1 && /15 "password"

or try a more basic password in case there's any weirdness with special characters

**Comment by MarkHofmann11** [comment]
Good idea, GP.  I just emailed Thaeland so we could try the PW without the
"!" in it.  Unix might not like a "!" in the password.

**Comment by Thaeland** [comment]
Changing the password fixed the problem. Thanks

**Comment by wwiv** [comment]
Strange. I just added a unit test to test the line:
```
@1234 &* "pass!"
```

And that parsed correctly.   Can you send me more details of the format of the line that didn't parse? I'd like to reproduce it and fix it.

**Comment by MarkHofmann11** [comment]
I'll send it to you, Rushfan.  Thanks!

**Comment by granitepenguin** [comment]
I don't remember exactly what I tried when I saw something similar with mine, but it could be a number of things: length, characters, upper/lower, punctuation.  I do remember my first password was auto-generated and ugly, and didn't work.  I found simple works better, hence the suggestion.  

Unfortunately, there's no documentation (besides the source code) about what constitutes a valid password, because the "old way" generated the password for you rather than you entering it manually.

It's also worth mentioning that I do have a "!" in mine, so just having one is not the issue.
